pkg/controllers/management/certsexpiration/certsexpiration.go

```diff
@@ -0,0 +1,76 @@
+package certsexpiration
+
+import (
+	"context"
+	"reflect"
+	"time"
+
+	"github.com/rancher/rancher/pkg/clustermanager"
+	"github.com/rancher/rancher/pkg/rkecerts"
+	rkeCluster "github.com/rancher/rke/cluster"
+	v1 "github.com/rancher/types/apis/core/v1"
+	v3 "github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// This controller handles cert expiration for local cluster only
+func Register(ctx context.Context, management *config.ManagementContext, clusterManager *clustermanager.Manager) {
+	c := &certsExpiration{
+		clusterClient:   management.Management.Clusters(""),
+		configMapLister: management.Core.ConfigMaps("kube-system").Controller().Lister(),
+	}
+	management.Management.Clusters("").AddHandler(ctx, "certificate-expiration", c.sync)
+}
+
+type certsExpiration struct {
+	clusterClient   v3.ClusterInterface
+	configMapLister v1.ConfigMapLister
+}
+
+func (c *certsExpiration) sync(key string, cluster *v3.Cluster) (runtime.Object, error) {
+	if cluster.Name != "local" {
```

Comment:
this can cause a panic, we need to make sure that cluster isn't nil first

Reply:
also a similar check should be added here i think https://github.com/rancher/rancher/blob/master/pkg/controllers/user/certsexpiration/certsexpiration.go#L47

Reply:
This one yes, but linked user one should be ok since it's checking `cluster == nil `, unless you wan to check `c.ClusterName` ? That code has been in prod for a long time so should be fine.

Reply:
thats cool then, thanks!